src/main/java/com/commercetools/sync/commons/utils/GraphQlQueryAll.java

```diff
@@ -0,0 +1,142 @@
+package com.commercetools.sync.commons.utils;
+
+import com.commercetools.sync.commons.helpers.GraphQlRequest;
+import com.commercetools.sync.commons.helpers.GraphQlResult;
+import com.commercetools.sync.commons.models.ResourceKeyId;
+import io.sphere.sdk.client.SphereClient;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+final class GraphQlQueryAll {
```

Comment:
This class only runs pagination only for the type ResourceKeyId, so not a generic implementation.
When we need another type of graph query we can not use it without modification, so making this generic could save us some time, also I would like to use this class for project sync with a generic implementation because we need a generic pagination implementation there too.

Here the decision should be based on the effort, we could check how much effort needed to make this generic and decide what to do.

Reply:
@ahmetoz To make this generic we need to make the result type generic. Also, we should make the utility 'queryAll()' method generic like Lam mentioned here: 

> queryAll() should be a util method that anyone can use to query for graphql, right? Comparing to other queryAll() in this class, this one for graphql is too restrictive. It already has a fixed query for you and you cannot change the query. Is it in the scope of this task to create a general queryAll()?

Reply:
Of course, for a generic implementation, you need generic types, but that is possible with inheritance. Quickly checked but it would require some effort to do it because of the deserialization need of the queries and their results.. 

Maybe we could open an issue for it and investigate how it could be done later. But opening an issue is required, instead of skipping this.

Reply:
https://github.com/commercetools/commercetools-sync-java/issues/627